### ISSUE TITLE
Scroll dungeon view

### ISSUE THREAD

**Issue by Mikolaj** [LambdaHack/LambdaHack#22]
After #113, #114 and #115 are done, let the different parts of a level be shown in the viewport. I'm inclined to center view continuously on the player leader position and on crosshair in aiming mode. The letter lets player view any part of the level, but the former doesn't burden the player and the UI with extra commands, scrollbars and tasks. 

The issue to decide is where to stop centering when the player is close to the edge of a level. Perhaps never stop centering, which reduced the visible level area but, e.g., makes multi-line messages less likely to obscure the leader. Also, it's then easier to locate the leader visually --- he's always exactly at the centre of the screen. However, if the whole level fits in the viewport, it's iffy to show just a quarter of it.

**Comment by Mikolaj** [comment]
An idea form other roguelikes: with Caps Lock, let movement commands scroll the dungeon view, not move actors.

**Comment by Mikolaj** [comment]
A nice summary of possible styles of scrolling: http://www.andrevdm.com/posts/multi/2018-04-02-haskell-rogue-like_09.html

**Comment by Mikolaj** [comment]
For now, let's postpone this and focus on dungeon levels that are equal or smaller to the viewport. When a game appears that needs huge levels and so needs this engine feature, let's work on it.
